system/security/shared/caching.cpp

```diff
@@ -627,7 +627,16 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
                     return true;
                 }
                 else
+                {
                     matchedRes = res;//allowed at this scope, but must also look at child scopes
+                }
+            }
+            else
+            {
+#ifdef _DEBUG
+                DBGLOG("FileScope %s for user %s managed but not cached", fullScope, sec_user.getName());
+#endif
+                return false;//force callout to LDAP
```

Comment:
Would it be better if you set the matchedRes = false and break so that you may let line 656-663 to handle this case.